### PR TITLE
Add ability to withdraw ERC20 tokens from Bridge

### PR DESCRIPTION
**Pull request by maptuhec**
* Added access control contract
* Withdraw function added to the gravity bridge contract

### CODE REVIEW

File: solidity/test/sendToCosmos.ts [hunk 1]
```diff
@@ -14,10 +16,13 @@ import {
 chai.use(solidity);
 const { expect } = chai;
 
+let bridgeAccessControl:any
```
Comment (Holymir) [inline solidity/test/sendToCosmos.ts:19]: you can declare the type of the variable as BridgeAccessControl

File: solidity/test/sendToCosmos.ts [hunk 2]
```diff
@@ -14,10 +16,13 @@ import {
...
 
 async function runTest(opts: {}) {
 
 
+  const BridgeAccessControl = await ethers.getContractFactory("BridgeAccessControl");
+  bridgeAccessControl = (await BridgeAccessControl.deploy()) as BridgeAccessControl;
```
Comment (Holymir) [inline solidity/test/sendToCosmos.ts]: You don't need to specify the contract type even if the variable is declared as 'any'

File: solidity/test/arbitrary-logic.ts [hunk 3]
```diff
@@ -35,6 +36,11 @@ async function runTest(opts: {
 
   // Prep and deploy contract
   // ========================
+  let bridgeAccessControl:any
```
Comment (Holymir) [inline solidity/test/arbitrary-logic.ts]: I think it is better to set the variable type as BridgeAccessControl here and omit "as BridgeAccessControl" on row 42

File: solidity/test/arbitrary-logic.ts [hunk 4]
```diff
@@ -295,7 +301,14 @@ describe("submitLogicCall tests", function () {
 
 // This test produces a hash for the contract which should match what is being used in the Go unit tests. It's here for
 // the use of anyone updating the Go tests.
-describe("logicCall Go test hash", function () {
+describe("logicCall Go test hash", async function () {
+  let bridgeAccessControl:any
+
+  beforeEach(async () => {
+    const BridgeAccessControl = await ethers.getContractFactory("BridgeAccessControl");
+    bridgeAccessControl = (await BridgeAccessControl.deploy()) as BridgeAccessControl;
```
Comment (Holymir) [inline solidity/test/arbitrary-logic.ts:309]: 'as BridgeAccessControl' You don't need that

File: solidity/test/constructor.ts [hunk 5]
```diff
@@ -15,7 +16,18 @@ chai.use(solidity);
 const { expect } = chai;
 
 describe("constructor tests", function() {
+
+
+  let bridgeAccessControl:any
```
Comment (Holymir) [inline solidity/test/constructor.ts:21]: I think it is better to set the variable type as BridgeAccessControl here and omit "as BridgeAccessControl" on row 25

File: solidity/test/deployERC20.ts [hunk 6]
```diff
@@ -17,7 +18,10 @@ const { expect } = chai;
 
 async function runTest(opts: {}) {
 
+  let bridgeAccessControl:any
```
Comment (Holymir) [inline solidity/test/deployERC20.ts:21]: I think it is better to set the variable type as BridgeAccessControl here and omit "as BridgeAccessControl" on row 24

File: solidity/test/gas-test.ts [hunk 7]
```diff
@@ -14,6 +15,14 @@ chai.use(solidity);
 const { expect } = chai;
 
 describe("Gas tests", function () {
+
+    let bridgeAccessControl:any
```
Comment (Holymir) [inline solidity/test/gas-test.ts:19]: I think it is better to set the variable type as BridgeAccessControl here and omit "as BridgeAccessControl" on row 23

File: solidity/test/happy-path.ts [hunk 8]
```diff
@@ -16,6 +17,15 @@ const { expect } = chai;
 
 
 describe("Gravity happy path valset update + batch submit", function () {
+
+  let bridgeAccessControl:any
```
Comment (Holymir) [inline solidity/test/happy-path.ts:21]: I think it is better to set the variable type as BridgeAccessControl here and omit "as BridgeAccessControl" on row 25

File: solidity/test/submitBatch-vs-logicCall.ts [hunk 9]
```diff
@@ -19,6 +21,8 @@ import { ReentrantERC20 } from "../typechain/ReentrantERC20";
 chai.use(solidity);
 const { expect } = chai;
 
+let bridgeAccessControl:any
```
Comment (Holymir) [inline solidity/test/submitBatch-vs-logicCall.ts:24]: I think it is better to set the variable type as BridgeAccessControl here and omit "as BridgeAccessControl" on row 65

File: solidity/test/submitBatch.ts [hunk 10]
```diff
@@ -14,6 +15,7 @@ import {
 
 chai.use(solidity);
 const { expect } = chai;
+let bridgeAccessControl:any
```
Comment (Holymir) [inline solidity/test/submitBatch.ts:18]: I think it is better to set the variable type as BridgeAccessControl here and omit "as BridgeAccessControl" on row 38

File: solidity/test/updateValset.ts [hunk 11]
```diff
@@ -29,7 +31,13 @@ async function runTest(opts: {
   notEnoughReward?: boolean;
   withReward?: boolean;
 }) {
+
+  let bridgeAccessControl:any
```
Comment (Holymir) [inline solidity/test/updateValset.ts:35]: I think it is better to set the variable type as BridgeAccessControl here and omit "as BridgeAccessControl" on row 38

File: solidity/test/withdrawERC20.ts [hunk 12]
```diff
@@ -0,0 +1,92 @@
...
+  examplePowers
+} from "../test-utils/pure";
+
+chai.use(solidity);
+const { expect } = chai;
+
+describe("Withdraw ERC20 Bridge Tests", function() {
+
+
+  let bridgeAccessControl:any
```
Comment (Holymir) [inline solidity/test/withdrawERC20.ts:20]: I think it is better to set the variable type as BridgeAccessControl here and omit "as BridgeAccessControl" on row 28

**Review by Holymir — APPROVED** [review]
In general you can set the type of bridgeAccessControl in the unit tests as a "BridgeAccessControl" and omit "as BridgeAccessControl" while deploying contract.

Even if you leave the type as "any" you don't need to  set it "as BridgeAccessControl" and you can remove the import for the BridgeAccessControl.

**Review by Holymir — APPROVED** [review]

**Review by kstoykov — APPROVED** [review]

**Pull request merged**